### PR TITLE
Restyled onboarding

### PR DESCRIPTION
**Pull request by nwalters512**
Rewrote the entire onboarding flow to be more modular and look more pretty. This breaks onboarding and subsequent data redownloads out of `LaunchActivity` and into separate activities (`OnboardingActivity` and `RedownloadAcitivty`, respectively). It also includes a more useful and integrated myTBA onboarding. Finally, this changes the regular myTBA sign in (the one users would see if they sign in after initial setup) to use a flow similar to that found in initial setup.

See the following screenshots for how the new onboarding flow looks and functions.

### CODE REVIEW

**Comment by fangeugene** [comment]
Awesome

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 language: android
 jdk: oraclejdk8
-sudo: false
```
Comment (phil-lopreiato) [inline .travis.yml]: Put this back. It causes travis builds to run on their newer container-based infrastructure

http://docs.travis-ci.com/user/workers/container-based-infrastructure/

File: android/src/main/java/com/thebluealliance/androidclient/activities/LaunchActivity.java [hunk 2]
```diff
@@ -92,227 +66,65 @@ protected void onCreate(Bundle savedInstanceState) {
...
-
-    @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        super.onSaveInstanceState(outState);
-        outState.putString(CURRENT_LOADING_MESSAGE_KEY, currentLoadingMessage);
     }
 
-    private boolean checkDataRedownload() {
+    private boolean checkDataRedownload(Intent intent) {
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
```
Comment (phil-lopreiato) [inline android/src/main/java/com/thebluealliance/androidclient/activities/LaunchActivity.java]: While you're at this, the `SharedPreference` instance should come from DI (the MyTbaModule)

File: android/src/main/java/com/thebluealliance/androidclient/activities/MyTBAOnboardingActivity.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+public class MyTBAOnboardingActivity extends AppCompatActivity implements View.OnClickListener, PlusManager.Callbacks, MyTBAOnboardingViewPager.Callbacks {
+
+    private static final String MYTBA_LOGIN_COMPLETE = "mytba_login_complete";
+
+    private MyTBAOnboardingViewPager mMyTBAOnboardingViewPager;
+    private TextView mContinueButtonText;
+
+    private boolean isMyTBALoginComplete = false;
+
+    private PlusManager mPlusManager;
```
Comment (phil-lopreiato) [inline android/src/main/java/com/thebluealliance/androidclient/activities/MyTBAOnboardingActivity.java]: DI

File: android/src/main/java/com/thebluealliance/androidclient/activities/OnboardingActivity.java [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+import com.thebluealliance.androidclient.accounts.PlusManager;
+import com.thebluealliance.androidclient.adapters.FirstLaunchPagerAdapter;
+import com.thebluealliance.androidclient.background.LoadTBADataTaskFragment;
+import com.thebluealliance.androidclient.background.firstlaunch.LoadTBAData;
+import com.thebluealliance.androidclient.helpers.ConnectionDetector;
+import com.thebluealliance.androidclient.views.DisableSwipeViewPager;
+import com.thebluealliance.androidclient.views.MyTBAOnboardingViewPager;
+
+/**
+ * Created by Nathan on 8/13/2015.
```
Comment (phil-lopreiato) [inline android/src/main/java/com/thebluealliance/androidclient/activities/OnboardingActivity.java]: Don't need

File: android/src/main/java/com/thebluealliance/androidclient/views/MyTBAOnboardingViewPager.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.thebluealliance.androidclient.R;
+import com.thebluealliance.androidclient.adapters.MyTBAOnboardingPagerAdapter;
+
+import me.relex.circleindicator.CircleIndicator;
+
+/**
+ * Created by Nathan on 8/16/2015.
```
Comment (phil-lopreiato) [inline android/src/main/java/com/thebluealliance/androidclient/views/MyTBAOnboardingViewPager.java]: Don't need

File: android/src/main/java/com/thebluealliance/androidclient/background/LoadTBADataTaskFragment.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        } else if (args != null) {
+            // Don't load any data
+        } else {
+            // Load all data
+            dataToLoad = new Short[]{LoadTBAData.LOAD_TEAMS, LoadTBAData.LOAD_EVENTS, LoadTBAData.LOAD_DISTRICTS};
+        }
+
+        if (task == null) {
+            task = new LoadTBAData(this, getActivity());
```
Comment (phil-lopreiato) [inline android/src/main/java/com/thebluealliance/androidclient/background/LoadTBADataTaskFragment.java]: This should become a service. Doesn't have to be now though

File: android/src/main/java/com/thebluealliance/androidclient/background/firstlaunch/LoadTBAData.java [hunk 7]
```diff
@@ -32,13 +32,18 @@
 /**
  * File created by phil on 4/20/14.
  */
-public class LoadAllData extends AsyncTask<Short, LoadAllData.LoadProgressInfo, Void> {
+public class LoadTBAData extends AsyncTask<Short, LoadTBAData.LoadProgressInfo, Void> {
```
Comment (phil-lopreiato) [inline android/src/main/java/com/thebluealliance/androidclient/background/firstlaunch/LoadTBAData.java]: Delete header, move to service?

**Comment by phil-lopreiato** [comment]
Love this!

File: .travis.yml (hunk 1 above)
Comment (nwalters512) [inline .travis.yml]: Funny story about this, I was reading about the new container stuff and thought "that looked cool!" so I made a new fork on my branch to add that, forgot to commit it, switched to this branch to add some stuff, realized I hadn't committed on the other branch, decided this was out of scope, and took it out again... tl;dr was trying to be helpful but it didn't work

File: android/src/main/java/com/thebluealliance/androidclient/background/LoadTBADataTaskFragment.java (hunk 6 above)
Comment (nwalters512) [inline android/src/main/java/com/thebluealliance/androidclient/background/LoadTBADataTaskFragment.java]: I figured that was out of scope for the work I was doing.

Comment (phil-lopreiato) [inline android/src/main/java/com/thebluealliance/androidclient/background/LoadTBADataTaskFragment.java]: Yeah, it can be the a future step. :+1:

**Pull request merged**